src/main/java/minijava/Cli.java

```diff
@@ -20,15 +21,34 @@
     private final PrintStream out;
     private final PrintStream err;
     private final JCommander jCommander;
+    /**
+     * Did the command argument parsing fail?
+     */
+    private final boolean invalidArguments;
 
 
     Cli(OutputStream out, OutputStream err, String... args) {
```

Comment:
I didn't really like it before, but now that we are adding more functionality... 

Constructors should not cause side effects. I would prefer to add a static factory method `static Cli create(OutputStream, OutputStream, String... args)` and make the constructor private. And maybe rethrow the `ParameterException` from that static factory method and handle it where the `Cli.create` method is called.